src/corelibs/U2Core/src/datatype/primers/PrimerValidator.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * UGENE - Integrated Bioinformatics Tools.
+ * Copyright (C) 2008-2021 UniPro <ugene@unipro.ru>
+ * http://ugene.net
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA 02110-1301, USA.
+ */
+
+#include "PrimerValidator.h"
+
+#include <U2Core/AppContext.h>
+#include <U2Core/U2AlphabetUtils.h>
+
+namespace U2 {
+
+PrimerValidator::PrimerValidator(QObject *parent, bool allowExtended)
+    : QRegExpValidator(parent) {
+    const DNAAlphabet *alphabet = AppContext::getDNAAlphabetRegistry()->findById(
+        allowExtended ? BaseDNAAlphabetIds::NUCL_DNA_EXTENDED() : BaseDNAAlphabetIds::NUCL_DNA_DEFAULT());
+    QByteArray alphabetChars = alphabet->getAlphabetChars(true);
```

Comment:
Do you really need both cases here? 
You run toUpper in validate(), so standard chars only are enough (both cases are used very rarely in our code)

Reply:
Don't get, what both cases are you talking about?
Just in case, this validator just has been moved from Primer.h/cpp files, there are no changes in there!

Reply:
It was about 'true' here: alphabet->getAlphabetChars(true);
Ok, I see this the old code. We can keep it as it is